src/dataflow/src/source/mod.rs

```diff
@@ -1167,9 +1171,16 @@ where
         });
     }
 
-    let (stream, capability) = source(scope, name, move |info| {
+    let (stream, capability) = source(scope, name.clone(), move |info| {
         let activator = Arc::new(scope.sync_activator_for(&info.address[..]));
 
+        let metrics_name = upstream_name.unwrap_or(name);
+        let mut metrics = SourceMetrics::new(&metrics_name, id, &worker_id.to_string(), logger);
+
+        if active {
+            metrics.add_partition(&PartitionId::File);
```

Comment:
Is this actually a file? Ideally we would track whatever source type this is precisely, if we can't or it doesn't matter (which it seems like it shouldn't here) we should just create an `Unknown` variant.

Reply:
Why do we actually track the type for non Kafka sources (which are the only ones that actually have partitions)? Looking at the existing ones we just report the name of the source. `PartitionId::S3` => `"s3"`, `PartitionId::Kinesis` => `"kinesis"` and `PartitionId::File` => `"0"`.

What do you think of defining PartitionId like so and folding all singular uses under it?

```rust
enum PartitionId {
    Kafka(i32),
    Single(&'static str),
}
```

Reply:
> Why do we actually track the type for non Kafka sources (which are the only ones that actually have partitions)? Looking at the existing ones we just report the name of the source. `PartitionId::S3` => `"s3"`, `PartitionId::Kinesis` => `"kinesis"` and `PartitionId::File` => `"0"`.
> 
> What do you think of defining PartitionId like so and folding all singular uses under it?
> 
> ```rust
> enum PartitionId {
>     Kafka(i32),
>     Single(&'static str),
> }
> ```

I like that idea a lot, fwiw.

Reply:
@quodlibetor @benesch I added a `PartitionId::None` option that I used for SimpleSources, I think this makes the most sense and we don't really need to put the name of the source in the partition id.

If this sounds good to both I'll make a follow up PR that will use `PartitionId::None` for all the sources that don't currently use partitions

Reply:
SGTM!